src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzFileAclStep.java

```diff
@@ -73,12 +74,24 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
             // Now, how to figure out if the failure is due to IAM propagation delay. We know it will
             // be a 400 - bad request and the docs indicate the reason will be "badRequest". So for now
             // we will log alot and retry on that.
-            if (ex.getCode() == 400 && (StringUtils.equals(ex.getReason(), "badRequest") ||
-                 ex.getMessage().contains("Could not find group"))) {
-                logger.info("Maybe caught an ACL propagation error: " + ex.getMessage()
-                    + " reason: " + ex.getReason(), ex);
+            // Note from DR-1760 - Potentially could remove this catch, should be handled with ApiException below
+            logger.error("[SnapshotACLException] StorageException, potentially an ACL propagation error. " +
+                "Message: {}, Reason: {}", ex.getMessage(), ex.getReason());
+            if (ex.getCode() == 400 && (StringUtils.equals(ex.getReason(), "badRequest"))) {
+                logger.error("[SnapshotACLException] Retrying! Bad Request.");
                 return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, ex);
             }
+            throw ex;
+        } catch (ApiException ex) {
```

Comment:
It looks like [ApiException](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/app/controller/exception/ApiException.java#L5) is for 500 errors. Is that what we expect here?

Reply:
Hm, I suppose so? The actual error we're looking for is nested in this exception. In the condition below, I check the root "cause" of the ApiException, and then see if matches the storage exception we're seeing.

Reply:
Oh I see! I think it makes sense to keep the other check for bad requests then.